### PR TITLE
feat(payment): INT-1650 Adding IIN field to vaulted instruments

### PR DESCRIPTION
**Pull request by ric-sanchez**
[INT-1650](https://jira.bigcommerce.com/browse/INT-1650)

## What?
Adding BIN Number for vaulted credit cards in the form of the issuer_identification_number (iin), already stored in the BigPay database.

## Why?
This iin value should be received by Checkout SDK strategy so it can be used in the Cardinal's BIN Detection step.

## Testing / Proof
image

@bigcommerce/checkout @bigcommerce/payments @bigcommerce-labs/intersys-integrations

### CODE REVIEW

File: src/payment/strategies/braintree/braintree-credit-card-payment-strategy.spec.ts [hunk 1]
```diff
@@ -152,6 +152,7 @@ describe('BraintreeCreditCardPaymentStrategy', () => {
                     methodId: 'braintree',
                     paymentData: {
                         instrumentId: 'my_instrument_id',
+                        iin: '123123'
```
Comment (josesonora) [inline src/payment/strategies/braintree/braintree-credit-card-payment-strategy.spec.ts]: please add a trailing comma

File: src/payment/strategies/square/square-payment-strategy-mock.ts [hunk 2]
```diff
@@ -42,6 +42,7 @@ export function getPayloadVaulted(): OrderRequestBody {
         payment:  {
             paymentData: {
                 instrumentId: 'string',
+                iin: 'string'
```
Comment (josesonora) [inline src/payment/strategies/square/square-payment-strategy-mock.ts]: please add a trailing comma

File: src/payment/strategies/braintree/braintree-credit-card-payment-strategy.spec.ts (hunk 1 above)
Comment (ric-sanchez) [reply src/payment/strategies/braintree/braintree-credit-card-payment-strategy.spec.ts]: Done 👍

File: src/payment/strategies/square/square-payment-strategy-mock.ts (hunk 2 above)
Comment (ric-sanchez) [reply src/payment/strategies/square/square-payment-strategy-mock.ts]: Done 👍

File: src/payment/payment.ts [hunk 3]
```diff
@@ -46,3 +48,16 @@ export interface CryptogramInstrument {
...
+    status: string;
+    vendor: string;
+    cavv: string;
+    eci: string;
+    xid: string;
+}
+
+export interface ThreeDSecureToken {
+    token: string;
+}
```
Comment (cbandam) [inline src/payment/payment.ts]: Missing End of Line

**Comment by cbandam** [comment]
@josesonora @ric-sanchez does this PR has dependency with #565 due to the new ThreeDS interfaces? if so we should add this dependency explicitly in the description.

**Comment by JoseBobadilla20** [comment]
💚 Tested

File: src/payment/payment.ts [hunk 4]
```diff
@@ -32,6 +32,8 @@ export interface VaultedInstrument {
     instrumentId: string;
     ccCvv?: string;
     ccNumber?: string;
+    iin: string;
+    threeDSecure?: ThreeDSecure | ThreeDSecureToken;
```
Comment (josesonora) [inline src/payment/payment.ts]: Changes related to `3DS` shouldn't be in this PR since they were already merged

Comment (ric-sanchez) [reply src/payment/payment.ts]: Done 👍

**Pull request closed without merging**